commons/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -367,8 +452,28 @@ private void startUpdatingCategoryList() {
         if (lastUpdater != null) {
             lastUpdater.cancel(true);
         }
+
+        if (methodAUpdater != null) {
+            methodAUpdater.cancel(true);
+        }
+
+
+        ArrayList<CategoryItem> itemList = new ArrayList<CategoryItem>(itemSet);
+
+        methodAUpdater = new MethodAUpdater();
         lastUpdater = new CategoriesUpdater();
+
         Utils.executeAsyncTask(lastUpdater, executor);
+        Utils.executeAsyncTask(methodAUpdater, executor);
+
+
+        categoriesAdapter.setItems(itemList);
```

Comment:
You have to call setItems() and notifyDataSetInvalidated() after the AsyncTasks onPostExecutes are called. It looks like you're calling this immediately after starting the AsyncTasks.